webroot/js/directives/sentence-and-translations.dir.js

```diff
@@ -146,14 +125,45 @@
             resetLists();
         });
 
+        $scope.$on('randomSentenceRequested', function(event, lang) {
+            var url = rootUrl + '/sentences/random';
+            if (lang) {
+                url += '/' + lang;
+            }
+            vm.inProgress = true;
+            $http.get(url).then(function(result) {
+                var sentence = result.data.random;
+                var directTranslations = sentence.translations[0];
+                var indirectTranslations = sentence.translations[1];
+                init(vm.userLanguages, sentence, directTranslations, indirectTranslations);
+                initMenu(false, sentence.menu);
+                vm.inProgress = false;
+            });
```

Comment:
You are not checking for errors here. I think you can at least reset `vm.inProgress` in case of errors, like that:

```js
$http.get(url).then(function(result) {
    //...
}).finally(function() {
    vm.inProgress = false;
});